regridding/qc.ipynb

```diff
@@ -75,241 +75,157 @@
-    "    var_id = src_ds.attrs[\"variable_id\"]\n",
-    "    assert get_var_id(src_ds) == var_id, \"Variable ID mismatch\"\n",
-    "    assert get_var_id(regrid_ds) == var_id, \"Variable ID mismatch\"\n",
-    "\n",
-    "    fig, axes = plt.subplots(1, 2, figsize=(15, 4))\n",
-    "    fig.suptitle(\n",
-    "        f\"Variable: {var_id}     Model: {src_ds.attrs['source_id']}     Scenario: {src_ds.attrs['experiment_id']}\"\n",
+    "src_params = [extract_params_from_src_filepath(fp) for fp in src_fps]\n",
+    "for p_name, p_str in zip(\n",
+    "    [\"model\", \"scenario\", \"frequency\", \"variable_id\"], [models, scenarios, freqs, vars]\n",
+    "):\n",
+    "    assert all(\n",
+    "        [params[p_name] in p_str for params in src_params]\n",
+    "    ), f\"Source files submitted for regridding contain values for the {p_name} parameter ({', '.join(list(set([params[p_name] for params in src_params])))}) which were not supplied for QC in this notebook ({p_str}).\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d38b8d0",
+   "metadata": {},
+   "source": [
+    "Ignore certain files based on results in slurm output files:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "35c11c0c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# check slurm files\n",
+    "fps_to_ignore = summarize_slurm_out_files(slurm_dir)\n",
+    "for fp in fps_to_ignore:\n",
+    "    if fp in src_fps:\n",
+    "        src_fps.remove(fp)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4510c876",
+   "metadata": {},
+   "source": [
+    "Now compare expected files to existing files and make sure values OK. This will open and check files in parallel and could take a while. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "a7bfd8a4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds_errors, value_errors = compare_expected_to_existing_and_check_values(\n",
+    "    regrid_dir,\n",
+    "    regrid_batch_dir,\n",
+    "    slurm_dir,\n",
+    "    vars,\n",
+    "    freqs,\n",
+    "    models,\n",
+    "    scenarios,\n",
+    "    fps_to_ignore,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "30fa767c",
+   "metadata": {},
+   "source": [
+    "Here is a summary of the errors:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "5c24b3be",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "QC process complete: 0 errors found.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# print summary messages\n",
+    "error_count = len(ds_errors) + len(value_errors)\n",
+    "print(f\"QC process complete: {error_count} errors found.\")\n",
+    "if len(ds_errors) > 0:\n",
+    "    print(\n",
+    "        f\"Errors in opening some datasets. {len(ds_errors)} files could not be opened. See {str(error_file)} for error log.\"\n",
```

Comment:
The QC notebook complained that `error_file` was not defined here and execution stopped. I got around this temporarily just by removing the reference to `error_file` here.